cpp/include/raft/spectral/eigen_solvers.cuh

```diff
@@ -49,28 +50,52 @@ struct lanczos_solver_t {
   {
   }
 
-  index_type_t solve_smallest_eigenvectors(
+  auto solve_smallest_eigenvectors(
     raft::resources const& handle,
     matrix::sparse_matrix_t<index_type_t, value_type_t, size_type_t> const& A,
     value_type_t* __restrict__ eigVals,
     value_type_t* __restrict__ eigVecs) const
+  {
+    auto csr_structure =
+      raft::make_device_compressed_structure_view<index_type_t, index_type_t, index_type_t>(
+        const_cast<index_type_t*>(A.row_offsets_),
+        const_cast<index_type_t*>(A.col_indices_),
```

Comment:
Must we do this instead of removing const-reference from the function signature?

Reply:
That is a very good question, and I don't immediately remember why I did it this way. Let me try to figure that out and either fix it or add an explanatory comment.

Reply:
Realized that I lifted this from the same logic [here](https://github.com/rapidsai/raft/blob/branch-25.04/cpp/include/raft/sparse/solver/lanczos.cuh#L87-L97). Looks like there were some const-correctness issues in the PR that introduced the new solver. I'll see if I can fix them now.

Reply:
We are performing similar casts throughout the codebase around `csr_matrix` objects. To fix it here and elsewhere, we'll need quite a few changes to `csr_matrix` to support a `const` version of the `view` method. How would we feel about doing that in a follow-up, since it will be a fairly large change in its own right?

Reply:
Created the follow-up issue [here](https://github.com/rapidsai/raft/issues/2585).